packages/embark-ui/src/services/storage.js

```diff
@@ -1,72 +1,59 @@
-export function addEditorTabs({file}) {
-  return new Promise(resolve => {
-    const editorTabs = findOrCreateEditorTabs();
-    editorTabs.forEach(f => f.active = false);
-    const alreadyAddedFile = editorTabs.find(f => f.name === file.name)
-    if (alreadyAddedFile) {
-      alreadyAddedFile.active = true;
-      alreadyAddedFile.content = file.content;
-    } else {
-      file.active = true;
-      editorTabs.push(file);
-    }
-    localStorage.setItem('editorTabs', JSON.stringify(editorTabs));
-    resolve({response: {data: editorTabs}});
-  });
-}
-
-export function fetchEditorTabs() {
-  return new Promise(resolve => {
-    resolve({response: {data: JSON.parse(localStorage.getItem('editorTabs'))}});
-  });
-}
-
-export function removeEditorTabs({file}) {
-  return new Promise(resolve => {
-    const editorTabs = findOrCreateEditorTabs();
-    const filtered = editorTabs.filter(value => value.name !== file.name);
-    if (file.active && filtered.length) {
-      filtered[0].active = true;
-    }
-    localStorage.setItem('editorTabs', JSON.stringify(filtered));
-    resolve({response: {data: filtered}});
-  });
-}
-
-export function saveCredentials({token, host}) {
+/* global localStorage */
+
+export async function addEditorTabs({file}) {
```

Comment:
It is a personal preference but I prefer having the return `Promise`, it makes things more obvious at least to me that it will return a promise.
The async keyword may hide it.

Being that said, I am fine like that too :)

Reply:
@PascalPrecht since it returned a promise when it was a normal function, I made it an async function like the rest. My guess is it returned a promise to be consistent with all the other exports of the same module.

@alaibe good point, I didn't think of that.  I lean strongly toward making a function `async` if it returns a promise anyway and doing so doesn't otherwise "double promise" the return value (that cold be the case when returning a promise that wraps some callback api). However, if it's hurting readability, happy to consider reverting those changes.

Reply:
@michaelsbradleyjr I have just made some test and the `async` can be removed from the function, the ui should still work.
If you want to check yourself, you can start cockpit editor and make sure the tabs still works (add tab, remove tab, reload page and still have tabs)